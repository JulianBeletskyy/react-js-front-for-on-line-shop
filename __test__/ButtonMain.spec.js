import Enzyme, { shallow } from 'enzyme'
import React from 'react'
import BtnMain from '../src/components/buttons/btn_main.js'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('Render', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallow(<BtnMain className="test" disabled={true} title="Carrinho" />)
    })

    it('render component', () => {
       expect(wrapper.length).toEqual(1)
    });

    it('render child component', () => {
       expect(wrapper.children().length).toEqual(1)
    });

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('disables the button element', () => {
        const buttonElement = wrapper.find('button');
        expect(buttonElement.props().disabled).toBe(true);
    });

    it('contains component exact button', () => {
        expect(wrapper.contains(<button className="btn btn-primary test" disabled={true} type="button">Carrinho</button>)).toBe(true);
    });
});