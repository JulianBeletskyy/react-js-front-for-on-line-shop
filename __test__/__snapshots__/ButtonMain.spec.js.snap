// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render renders correctly 1`] = `
<div>
  <button
    className="btn btn-primary test"
    disabled={true}
    type="button"
  >
    Carrinho
  </button>
</div>
`;
